HOW/tech/applications/cell-joining-code/cell-joining-code-v1.md

```diff
@@ -0,0 +1,110 @@
+---
+HOW: /tech/applications/cell-joining-code
+Source: /social/process/how/how-v1
+Status: Draft
+Created: 2024-11-23
+Version: idx/1
+---
+
+## Title
+
+Cell Joining Code
+
+## Authors
+
+@mattyg <matt@buildyourweb.app>, @zippy <eric@holochain.org>, Claude AI 3.5 Sonnet
+
+## Abstract
+
+This HOW proposes a standardized URI format for cell joining codes that enables conductors to join specific networks. The format encapsulates all necessary information for network joining in a single, shareable URI string, excluding the actual DNA.  The format also allows app-specific data to be carried so that UIs can display arbitrary data as needed for good 
+
+## Motivation
+
+Joining a Cell network requires multiple pieces of information to be shared: DNA hash, network seed, membrane proof, and app-specific metadata. Several apps have run into the need to share this information as a single string and implemented their own joining code format.
+
+A standardized joining code format would:
+- Simplify the network joining process
+- Simplify use of network joining codes in apps
+- Reduce user error during configuration
+- Enable one-click joining capabilities
+- Facilitate easier sharing of network access
+- Facilitate services providing "always-on" Cells: simply provide the DNA and joining code
+
+## Rationale
+
+This specification meets the criteria of the motivation because:
+- it uses the standard URI format which makes it amenable to deep-linking
+- it uses a specific path in the URI format leaving other paths available under the `hc://` scheme for other uses.
+- it includes a well-known pattern for specifying attributes and thus arbitrary application-specific components, as well as optional data elements.
+
+## Technical Specification
+
+The cell joining URI format MUST follow this structure:
+
+```
+hc://join/<dna_hash>/<dna_modifiers>?proof=<membrane_proof>&metadata=<metadata>
+```
+
+Where:
+
+- The scheme MUST be `hc://`
+- The path MUST start with `join/` 
```

Comment:
Rationale: 
- a "verb" of what the code is intended to be used for
- makes the code self-describing
- leaves open the possibility for other kinds of codes with other verbs. Can we think of any or is this not expected?